src/com/lftechnology/java/training/niraj/ems/utils/DbFacade.java

```diff
@@ -0,0 +1,372 @@
+    }
+
+    /**
+     * Create update statement
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param connection
+     * @param table
+     * @param values
+     * @return
+     * @throws SQLException
+     */
+    public static PreparedStatement createUpdateStatement(Connection connection, String table, Map<String, String> values)
+            throws SQLException {
+        PreparedStatement stmt = connection.prepareStatement(addUpdateClause(table, values));
+        return setStatementParameters(stmt, values);
+    }
+
+    /**
+     * Creates update statement with conditions
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param connection
+     * @param table
+     * @param values
+     * @param conditions
+     * @return
+     * @throws SQLException
+     */
+    public static PreparedStatement createUpdateStatement(Connection connection, String table, Map<String, String> values,
+            Map<String, String> conditions) throws SQLException {
+        StringBuilder query = new StringBuilder();
+        if (values.containsKey("id")) {
+            values.remove("id");
+        }
+        query.append(addUpdateClause(table, values));
+        query.append(addWhereClause(conditions, Operators.AND));
+        PreparedStatement stmt = connection.prepareStatement(query.toString());
+        return setStatementParametersWithWhereClause(stmt, values, conditions);
+
+    }
+
+    /**
+     * Creates update statement with conditions
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param connection
+     * @param table
+     * @param values
+     * @param conditions
+     * @return
+     * @throws SQLException
+     */
+    public static PreparedStatement createUpdateStatement(Connection connection, String table, Map<String, String> values,
+            Map<String, String> conditions, Operators operator) throws SQLException {
+        StringBuilder query = new StringBuilder();
+        if (values.containsKey("id")) {
+            values.remove("id");
+        }
+        query.append(addUpdateClause(table, values));
+        query.append(addWhereClause(conditions, operator));
+        PreparedStatement stmt = connection.prepareStatement(query.toString());
+        return setStatementParametersWithWhereClause(stmt, values, conditions);
+
+    }
+
+    /**
+     * Adds update clause
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param table
+     * @param values
+     * @return
+     */
+    private static String addUpdateClause(String table, Map<String, String> values) {
+        StringBuilder query = new StringBuilder();
+        query.append("UPDATE ");
+        query.append(table);
+        query.append(" SET ");
+        for (Entry<String, String> entry : values.entrySet()) {
+            String key = entry.getKey();
+            if (!key.equals("id")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "id" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)